modules/percolator/src/main/java/org/elasticsearch/percolator/TransportPercolateAction.java

```diff
@@ -184,6 +184,9 @@ public static SearchRequest createSearchRequest(PercolateRequest percolateReques
                         }
                     } else if (token.isValue()) {
                         if ("size".equals(currentFieldName)) {
+                            if (percolateRequest.onlyCount()) {
```

Comment:
With strict duplicate checks enabled we would throw an error because `size` is specified twice (once by the user, once added internally by our implementation of `TransportPercolateAction`) and this error message gives the user a clear feedback to the user what is wrong instead of just giving a (technically correct but) misleading error to the user about a "duplicate field size".